File: GotlandFrontend/lib/model/day_booking.dart
import 'package:frontend_shared/util.dart' show ValueConverter;

import 'json_field.dart';

class DayBooking {
  DayBooking(this.id, this.firstName, this.lastName, this.email, this.phoneNo, this.dob, this.food, this.type, this.paymentConfirmed, this.created);

  final String id;
  final String firstName;
  final String lastName;
  final String email;
  final String phoneNo;
  final String dob;
  final String food;
  final String type;
  final bool paymentConfirmed;
  final DateTime created;

  factory DayBooking.fromMap(Map<String, dynamic> json) =>
      DayBooking(json[ID], json[FIRSTNAME], json[LASTNAME], json[EMAIL], json[PHONE_NO], json[DOB], json[FOOD], json[TYPE_ID], json[PAYMENT_CONFIRMED], ValueConverter.parseDateTime(json[CREATED]));
}

File: GotlandBackend/Models/DayBooking.cs
﻿using System;

namespace Accidis.Gotland.WebService.Models
{
	public sealed class DayBooking
	{
		public Guid Id { get; set; }
		public Guid EventId { get; set; }
		public string Reference { get; set; }
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string Email { get; set; }
		public string PhoneNo { get; set; }
		public string Gender { get; set; }
		public string Dob { get; set; }
		public string Food { get; set; }
		public Guid TypeId { get; set; }
		public bool PaymentConfirmed { get; set; }
		public DateTime? ConfirmationSent { get; set; }
		public DateTime Created { get; set; }
		public DateTime Updated { get; set; }
	}
}
